k8s-direct/build.rs

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Open Network Fabric Authors
+
+use std::fs;
+use std::path::PathBuf;
+
+fn get_agent_crd_url() -> String {
+    println!("cargo:rerun-if-changed=scripts/k8s-crd.env");
+
+    let env_file = dotenvy::from_filename_iter("scripts/k8s-crd.env")
+        .expect("Failed to read scripts/k8s-crd.env");
+
+    env_file
+        .filter_map(Result::ok)
+        .find_map(|(key, value)| {
+            if key == "K8S_GATEWAY_AGENT_CRD_URL" {
+                Some(value)
+            } else {
+                None
+            }
+        })
+        .expect("K8S_GATEWAY_AGENT_CRD_URL not found in scripts/k8s-crd.env")
+}
+
+fn fetch_crd(url: &str) -> String {
+    println!("cargo:note=Fetching CRD from: {url}");
+    ureq::get(url)
+        .call()
+        .expect("Failed to fetch agent CRD from url")
+        .body_mut()
+        .read_to_string()
+        .expect("Failed to read response body")
+}
+
+// FIXME(mvachhar): When kopium has a library mode, or when artifacts are no longer experimental in cargo,
+// use one of those methods to get rid of this function and its use.
+//
+// In the meantime, this is the best I could come up with that didn't require a different build command.
+fn install_kopium() -> PathBuf {
+    // Check if kopium binary exists, install if not
+    let kopium_path = PathBuf::from("./bin/kopium");
+    if !kopium_path.exists() {
+        println!("cargo:note=kopium binary not found, installing...");
+
+        // Create bin directory if it doesn't exist
+        let bin_dir = PathBuf::from("./bin");
+        fs::create_dir_all(&bin_dir).expect("Failed to create bin directory");
+
+        // Install kopium using cargo install
+        let output = std::process::Command::new("cargo")
+            .args(["install", "kopium", "--no-track", "--root", "."])
+            .output()
+            .expect("Failed to execute cargo install kopium");
```

Comment:
```suggestion
fn install_kopium() -> PathBuf {
    let out_dir = std::env::var("OUT_DIR").expect("OUT_DIR not set");
    let bin_dir = PathBuf::from(&out_dir).join("bin");
    let kopium_path = bin_dir.join("kopium");

    // Check if kopium binary exists, install if not
    if !kopium_path.exists() {
        println!("cargo:note=kopium binary not found, installing...");

        // Create bin directory if it doesn't exist
        fs::create_dir_all(&bin_dir).expect("Failed to create bin directory");

        // Install kopium using cargo install
        let output = std::process::Command::new("cargo")
            .args(["install", "kopium", "--no-track", "--root", &out_dir])
            .output()
            .expect("Failed to execute cargo install kopium");

```

Using `OUT_DIR`:
- uses the same directory as for the other artefacts (for example `target/x86_64-unknown-linux-gnu/debug/build/dataplane-k8s-direct-c0fbecd7fb0d98fb/out/bin/kopium`)
- avoids “polluting” the source tree, and the need for a local `.gitignore`
- makes it work with `cargo clean`, which wouldn't remove kopium otherwise.

Reply:
Fixed, `kopium` is now in `dpkd-sys`